cookbook/security/custom_authentication_provider.rst

```diff
@@ -218,12 +219,12 @@ the ``PasswordDigest`` header value matches with the user's password.
     class WsseProvider implements AuthenticationProviderInterface
     {
         private $userProvider;
-        private $cacheDir;
+        private $cachePool;
 
-        public function __construct(UserProviderInterface $userProvider, $cacheDir)
+        public function __construct(UserProviderInterface $userProvider, CacheItemPoolInterface $cachePool)
```

Comment:
AdapterInterface extends the PSR ones, so no problem here and indeed: Let's be as generic as possible